backend/src/nodes/image_util_nodes.py

```diff
@@ -12,110 +12,112 @@
 from .node_factory import NodeFactory
 from .properties.inputs import *
 from .properties.outputs import *
-from .utils.image_utils import alpha_overlay, calculate_ssim
+from .utils.image_utils import blend_images, calculate_ssim
 from .utils.pil_utils import *
 from .utils.utils import get_h_w_c
 
 
-@NodeFactory.register("chainner:image:overlay")
-class ImOverlay(NodeBase):
-    """OpenCV transparency overlay node"""
+@NodeFactory.register("chainner:image:blend")
+class ImBlend(NodeBase):
+    """Blending mode node"""
 
     def __init__(self):
         """Constructor"""
         super().__init__()
-        self.description = "Overlay transparent images on base image."
+        self.description = """Blends overlay image onto base image using 
+            specified mode and opacities."""
         self.inputs = [
-            ImageInput("Base").with_id(0),
+            ImageInput("Base Layer"),
             SliderInput(
                 "Base Opacity",
                 maximum=100,
                 default=100,
                 step=0.1,
                 controls_step=1,
                 unit="%",
-            ).with_id(5),
-            ImageInput("Overlay A").with_id(1),
-            SliderInput(
-                "Opacity A",
-                maximum=100,
-                default=50,
-                step=0.1,
-                controls_step=1,
-                unit="%",
             ).with_id(2),
-            ImageInput("Overlay B").make_optional().with_id(3),
+            ImageInput("Overlay Layer").with_id(1),
             SliderInput(
-                "Opacity B",
+                "Overlay Opacity",
                 maximum=100,
-                default=50,
+                default=100,
                 step=0.1,
                 controls_step=1,
                 unit="%",
-            ).with_id(4),
+            ),
+            BlendModeDropdown(),
         ]
         self.outputs = [ImageOutput()]
         self.category = IMAGE_UTILITY
-        self.name = "Overlay Images"
+        self.name = "Blend Images"
         self.icon = "BsLayersHalf"
         self.sub = "Compositing"
 
     def run(
         self,
         base: np.ndarray,
         opbase: float,
-        ov1: np.ndarray,
-        op1: float,
-        ov2: Union[np.ndarray, None],
-        op2: float,
+        ov: np.ndarray,
+        op: float,
+        blend_mode: int,
     ) -> np.ndarray:
-        """Overlay transparent images on base image"""
+        """Blend images together"""
 
         # Convert to 0.0-1.0 range
         opbase /= 100
-        op1 /= 100
-        op2 /= 100
+        op /= 100
 
         imgs = []
-        max_h, max_w = 0, 0
-        for img in base, ov1, ov2:
-            if img is not None:
-                h, w, c = get_h_w_c(img)
-                max_h = max(h, max_h)
-                max_w = max(w, max_w)
+        max_h = max_w = 0
+        for img in (base, ov):
```

Comment:
The loop seems like overkill for 2 images.

Reply:
I mostly just wanted to avoid doubling these lines:
```h, w, c = get_h_w_c(img)
max_h = max(h, max_h)
max_w = max(w, max_w)

# All inputs must be BGRA for alpha compositing to work
assert c != 4, f"Number of channels ({c}) unexpected"
if c == 1:
    img = cv2.cvtColor(img, cv2.COLOR_GRAY2BGRA)
elif c == 3:
    img = cv2.cvtColor(img, cv2.COLOR_BGR2BGRA)```

Reply:
I can change it if the extra smidge of performance overhead is worth the extra clutter, but it just seemed more readable to leave it as a loop.

Reply:
> the extra smidge of performance overhead

I don't care about the performance overhead of a single loop when the numpy stuff we're doing right after takes like 1000x longer. It's not worth worrying about.

What I meant is that the loop is overcomplicating things. The whole

```py
# All inputs must be BGRA for alpha compositing to work
assert c != 4, f"Number of channels ({c}) unexpected"
if c == 1:
    img = cv2.cvtColor(img, cv2.COLOR_GRAY2BGRA)
elif c == 3:
    img = cv2.cvtColor(img, cv2.COLOR_BGR2BGRA)
```

 should really be its own function and getting the max of the sizes of 2 images shouldn't be too hard.